Getting this error even with Pro subscription and Readding auth - Even With pro Account

Bug Description
Getting this error even with Pro subscription and Readding auth - Even With pro Account, IDE is fully functional 
It was working initially a few days ago.

[2026-01-08T14:10:10.772Z] [WARN] [CloudCode] Error at https://cloudcode-pa.googleapis.com: 429 - {
"error": {
"code": 429,
"message": "Resource has been exhausted (e.g. check quota).",
"status": "RESOURCE_EXHAUSTED"
}
}

[2026-01-08T14:10:10.772Z] [WARN] [CloudCode] All endpoints rate-limited for XXX@gmail.com
[2026-01-08T14:10:10.773Z] [WARN] [AccountManager] Rate limited: XXX@gmail.com. Available in 1m0s
[2026-01-08T14:10:10.773Z] [INFO] [CloudCode] Account XXX@gmail.com rate-limited, trying next...
[2026-01-08T14:10:10.773Z] [INFO] [AccountManager] Waiting 59s for sticky account: XXX@gmail.com
[2026-01-08T14:10:10.773Z] [INFO] [CloudCode] Waiting 59s for sticky account...
[2026-01-08T14:11:10.771Z] [INFO] [AccountManager] Shortest wait: 0s (account: XXX@gmail.com)
[2026-01-08T14:11:10.772Z] [WARN] [CloudCode] All 1 account(s) rate-limited. Waiting 0s...
[2026-01-08T14:11:10.773Z] [SUCCESS] [AccountManager] Rate limit expired for: XXX@gmail.com
[2026-01-08T14:11:10.774Z] [WARN] [CloudCode] All 1 account(s) rate-limited. Waiting 1m0s...
[2026-01-08T14:11:10.775Z] [ERROR] [API] Error: Error: No accounts available
at sendMessage (file:///Users/XXX/antigravity-proxy-cli/antigravity-claude-proxy/src/cloudcode/message-handler.js:95:23)
at async file:///Users/XXX/antigravity-proxy-cli/antigravity-claude-proxy/src/server.js:505:30
[2026-01-08T14:11:10.777Z] [WARN] [API] Returning error response: 500 api_error - No accounts available
[2026-01-08T14:12:10.772Z] [ERROR] [API] Error: Error: No accounts available
at sendMessage (file:///Users/XXX/antigravity-proxy-cli/antigravity-claude-proxy/src/cloudcode/message-handler.js:95:23)
at async file:///Users/XXX/antigravity-proxy-cli/antigravity-claude-proxy/src/server.js:505:30
[2026-01-08T14:12:10.773Z] [WARN] [API] Returning error response: 500 api_error - No accounts available

--- a/src/cloudcode/message-handler.js
+++ b/src/cloudcode/message-handler.js
@@ -72,8 +72,19 @@ export async function sendMessage(anthropicRequest, accountManager, fallbackEnab
                 const accountCount = accountManager.getAccountCount();
                 logger.warn(`[CloudCode] All ${accountCount} account(s) rate-limited. Waiting ${formatDuration(allWaitMs)}...`);
                 await sleep(allWaitMs);
+
+                // Add small buffer after waiting to ensure rate limits have truly expired
+                await sleep(500);
                 accountManager.clearExpiredLimits();
                 account = accountManager.pickNext(model);
+
+                // If still no account after waiting, try optimistic reset
+                // This handles cases where the API rate limit is transient
+                if (!account) {
+                    logger.warn('[CloudCode] No account available after wait, attempting optimistic reset...');
+                    accountManager.resetAllRateLimits();
+                    account = accountManager.pickNext(model);
+                }
             }
 
             if (!account) {
@@ -197,10 +208,10 @@ export async function sendMessage(anthropicRequest, accountManager, fallbackEnab
             }
 
             if (isNetworkError(error)) {
-                 logger.warn(`[CloudCode] Network error for ${account.email}, trying next account... (${error.message})`);
-                 await sleep(1000); // Brief pause before retry
-                 accountManager.pickNext(model); // Advance to next account
-                 continue;
+                logger.warn(`[CloudCode] Network error for ${account.email}, trying next account... (${error.message})`);
+                await sleep(1000); // Brief pause before retry
+                accountManager.pickNext(model); // Advance to next account
+                continue;
             }
 
             throw error;

--- a/src/cloudcode/streaming-handler.js
+++ b/src/cloudcode/streaming-handler.js
@@ -71,8 +71,19 @@ export async function* sendMessageStream(anthropicRequest, accountManager, fallb
                 const accountCount = accountManager.getAccountCount();
                 logger.warn(`[CloudCode] All ${accountCount} account(s) rate-limited. Waiting ${formatDuration(allWaitMs)}...`);
                 await sleep(allWaitMs);
+
+                // Add small buffer after waiting to ensure rate limits have truly expired
+                await sleep(500);
                 accountManager.clearExpiredLimits();
                 account = accountManager.pickNext(model);
+
+                // If still no account after waiting, try optimistic reset
+                // This handles cases where the API rate limit is transient
+                if (!account) {
+                    logger.warn('[CloudCode] No account available after wait, attempting optimistic reset...');
+                    accountManager.resetAllRateLimits();
+                    account = accountManager.pickNext(model);
+                }
             }
 
             if (!account) {
@@ -264,10 +275,10 @@ export async function* sendMessageStream(anthropicRequest, accountManager, fallb
             }
 
             if (isNetworkError(error)) {
-                 logger.warn(`[CloudCode] Network error for ${account.email} (stream), trying next account... (${error.message})`);
-                 await sleep(1000); // Brief pause before retry
-                 accountManager.pickNext(model); // Advance to next account
-                 continue;
+                logger.warn(`[CloudCode] Network error for ${account.email} (stream), trying next account... (${error.message})`);
+                await sleep(1000); // Brief pause before retry
+                accountManager.pickNext(model); // Advance to next account
+                continue;
             }
 
             throw error;

